nat/traversal/nat_proxy.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package traversal
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	log "github.com/cihub/seelog"
+	"github.com/mysteriumnetwork/node/services"
+)
+
+const logPrefix = "[NATProxy] "
+
+// NATProxy provides traffic proxying functionality for registered services
+type NATProxy struct {
+	servicePorts map[services.ServiceType]int
+}
+
+// NewNATProxy constructs an instance of NATProxy
+func NewNATProxy() *NATProxy {
+	return &NATProxy{
+		servicePorts: make(map[services.ServiceType]int),
+	}
+}
+
+// handOff traffic incoming through NATPinger punched hole should be handed off to NATPoxy
+func (np *NATProxy) handOff(serviceType services.ServiceType, incomingConn *net.UDPConn) {
+	proxyConn, err := np.getConnection(serviceType)
+	if err != nil {
+		log.Error(logPrefix, "failed to connect to NATProxy: ", err)
```

Comment:
Yes, basically theres no point to proceed. As for @donce, we don't return error here, since we run in throw away go routine in case of failure, just print an error for each connect attempt.